### PR TITLE
Add nodeCount to formatter

### PR DESCRIPTION
**Pull request by dramich**
Problem:
The CLI displays a node count for clusters which required the CLI to make multiple API calls

Solution:
Add a new nodeCount field based off a lister and included in the formatter to help the CLI 

Depends on https://github.com/rancher/types/pull/1133

Issue https://github.com/rancher/rancher/issues/25227

### CODE REVIEW

File: pkg/api/customization/cluster/formatter.go [hunk 1]
```diff
@@ -145,6 +148,12 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 		}
 	}
 
+	nodes, err := f.nodeLister.List(resource.ID, labels.Everything())
```
Comment (prachidamle) [inline pkg/api/customization/cluster/formatter.go:151]: resource.ID will the clusterID, does it also act as the namespaceID for the List call?

Comment (dramich) [reply pkg/api/customization/cluster/formatter.go:151]: Yes, the v3.nodes live in the cluster namespace

File: pkg/api/customization/cluster/formatter.go [hunk 2]
```diff
@@ -145,6 +148,12 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
...
+	if err != nil {
+		logrus.Warnf("error getting node list for cluster %s: %s", resource.ID, err)
+	} else {
+		resource.Values["nodeCount"] = len(nodes)
```
Comment (prachidamle) [inline pkg/api/customization/cluster/formatter.go:155]: We will be showing the nodeCount, irrespective of the cluster state; Also we list all nodes irrespective of the node status. Just confirming we want is this way? Also,  can we add an integration test to check the nodeCount of a cluster created?

Comment (dramich) [reply pkg/api/customization/cluster/formatter.go:155]: That is the current behavior when using the CLI

**Pull request merged**